Ago-Dic-2020/rodriguez-martinez-diego-jose/Practica 1/matrix.py

```diff
@@ -0,0 +1,27 @@
+class Matrix:
+    matrix_List = []
+
+    def __init__(self, matrix_string):
+        #Creo una lista de listas a partir del matrix_string; e es cada elemento de matrix_string.split 
+        # (ej:'9 8 7'), e.split() es una lista hecha de cada iteracion de e (ej: [9,8,7])
+        #
+
+        self.matrix_List = [[int(num) for num in e.split()] for e in matrix_string.split("\n")] 
+
+    def row(self, index):
+        #debido a la forma en la que construi la matriz, solo es cuestion de regresar el item de la lista
+        #de listas que corresponde a index
+
+        return self.matrix_List[index-1]
+
+    def column(self, index):
+        #List comprehension que te regresa la columna que corresponde al indice; La idea es recorrer las rows
+        # mientras que la columna se queda fija en el valor del index  
+
+        return [self.matrix_List[i][index-1] for i in range( len(self.matrix_List)) ]
```

Comment:
Hay un espacio de más entre `range(` y `len(`

Reply:
Incluso hay un approach más sencillo en el que puedes recorrer las filas de la matriz con el `for` comprimido, y acceder a la respectiva columna desde el row que estás iterando 😄

Reply:
Tiene razon, aun me sigue revolviendo a veces eso que el for en python es mas como un for each.